### ISSUE TITLE
Integration proposal - ClamAv antivirus

### ISSUE THREAD

**Issue by csharpfritz** [CommunityToolkit/Aspire#102]
### Overview

I've started an integration sample that uses [ClamAv anti-malware](https://www.clamav.net) to scan file uploads for malware.  Source at:  https://github.com/csharpfritz/aspireantivirus

Would the team be interested in a contribution of that initial sample code and formalizing the feature?

### Usage example

This integration could be used to scan uploads to a service to ensure that no malware is included

### Help us help you

Yes, I'd like to be assigned to work on this item

**Comment by aaronpowell** [comment]
AV to scan user provided files makes a lot of sense and you're making me question why I've never done that in a project before 🤣.

Would be happy to have it as a contribution, I notice your repo has a hosting integration only and the client is more directly using the client for ClamAV rather than an Aspire client integration. Would it make sense to also include a client integration to simplify the workflow of consuming it?

**Comment by csharpfritz** [comment]
You're right, we should put together a client implementation as well... and I started one, then got a little confused.  I could use some assistance in tightening that client implementation, and would be happy to share the work I started.

**Comment by aaronpowell** [comment]
If you want to get a pull request up @csharpfritz we can help iterate and refine it

**Comment by dansiegel** [comment]
> AV to scan user provided files makes a lot of sense and you're making me question why I've never done that in a project before 🤣.

I'm with @aaronpowell... I would LOVE to see this and happy to lend a hand if needed

**Comment by csharpfritz** [comment]
I have some content built and I can share it to the repoOn Nov 15, 2024, at 15:53, Dan Siegel ***@***.***> wrote:﻿Re: [CommunityToolkit/Aspire] Integration proposal - nClam antivirus (Issue #102)AV to scan user provided files makes a lot of sense and you're making me question why I've never done that in a project before 🤣.I'm with @aaronpowell... I would LOVE to see this and happy to lend a hand if needed—Reply to this email directly, view it on GitHub or unsubscribe.You are receiving this email because you commented on the thread.Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by csharpfritz** [comment]
I'll put together that PR with the nClam support this week

**Comment by csharpfritz** [comment]
I'm submitting a PR for this now
